### ISSUE TITLE
Nextcloud samsung S23 imediate error on opening

### ISSUE THREAD

**Issue by sjabbie888** [nextcloud/android#11389]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

Samsung s23
Installation and reinstallation of nextcloud
Immediate failure on opening nextcloud after log-in with Filerun credentials of private server (issue not existant on S10 and S20

### Expected behaviour

Not crash

### Actual behaviour

Crash:

### Cause of error
```java
Exception in thread "Thread-10" java.lang.StringIndexOutOfBoundsException: begin 0, end 8, length 5
    at java.lang.String.checkBoundsBeginEnd(String.java:3942)
    at java.lang.String.substring(String.java:2467)
    at com.owncloud.android.datamodel.OCFile.getLocalId(OCFile.java:606)
    at com.owncloud.android.datamodel.FileDataStorageManager.createContentValuesBase(FileDataStorageManager.java:461)
    at com.owncloud.android.datamodel.FileDataStorageManager.createContentValuesForFolder(FileDataStorageManager.java:478)
    at com.owncloud.android.datamodel.FileDataStorageManager.saveFolder(FileDataStorageManager.java:392)
    at com.owncloud.android.operations.RefreshFolderOperation.synchronizeData(RefreshFolderOperation.java:538)
    at com.owncloud.android.operations.RefreshFolderOperation.fetchAndSyncRemoteFolder(RefreshFolderOperation.java:409)
    at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:239)
    at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:399)
    at java.lang.Thread.run(Thread.java:1012)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30240090`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `dm1q`
* Model: `SM-S911B`
* Id: `TP1A.220624.014`
* Product: `dm1qxeea`

### Firmware
* SDK: `33`
* Release: `13`
* Incremental: `S911BXXU1AWA6`

### Android version

13 ui 5.1

### Device brand and model

Samsung s23

### Stock or custom OS?

Stock

### Nextcloud android app version

30240090

### Nextcloud server version

Filerun

### Using a reverse proxy?

No

### Cause of error
```java
Exception in thread "Thread-10" java.lang.StringIndexOutOfBoundsException: begin 0, end 8, length 5
    at java.lang.String.checkBoundsBeginEnd(String.java:3942)
    at java.lang.String.substring(String.java:2467)
    at com.owncloud.android.datamodel.OCFile.getLocalId(OCFile.java:606)
    at com.owncloud.android.datamodel.FileDataStorageManager.createContentValuesBase(FileDataStorageManager.java:461)
    at com.owncloud.android.datamodel.FileDataStorageManager.createContentValuesForFolder(FileDataStorageManager.java:478)
    at com.owncloud.android.datamodel.FileDataStorageManager.saveFolder(FileDataStorageManager.java:392)
    at com.owncloud.android.operations.RefreshFolderOperation.synchronizeData(RefreshFolderOperation.java:538)
    at com.owncloud.android.operations.RefreshFolderOperation.fetchAndSyncRemoteFolder(RefreshFolderOperation.java:409)
    at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:239)
    at com.owncloud.android.lib.common.operations.RemoteOperation.run(RemoteOperation.java:399)
    at java.lang.Thread.run(Thread.java:1012)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30240090`
* Build flavor: `gplay`

### Device information
* Brand: `samsung`
* Device: `dm1q`
* Model: `SM-S911B`
* Id: `TP1A.220624.014`
* Product: `dm1qxeea`

### Firmware
* SDK: `33`
* Release: `13`
* Incremental: `S911BXXU1AWA6`

### Server error logs

```bash
Na
```

### Additional information

Na

**Comment by AlvaroBrey** [comment]
Fixed by #11391

**Comment by sjabbie888** [comment]
[#11391] Dev Version Fixes Issue.

Stable-3.24 after removal and re-installation (and granting Rights) still continous with the above behaviour in my case.
